pkg/operator/upgradebackupcontroller/upgradebackupcontroller.go

```diff
@@ -0,0 +1,288 @@
+			Message: fmt.Sprintf("Upgrade backup pod created on node %s", backupNodeName),
+		}, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	// Check the existing Pod Status and update condition respectively.
+	// PodSucceeded => ConditionTrue
+	// PodFailed =>  ConditionFalse
+	// everything else ConditionUnknown
+	switch {
+	case backupPod.Status.Phase == corev1.PodSucceeded:
+		backupCondition := v1helpers.FindOperatorCondition(operatorStatus.Conditions, backupConditionType)
+		if backupCondition.Reason != backupSuccess {
+			return &operatorv1.OperatorCondition{
+				Status:  operatorv1.ConditionTrue,
+				Type:    backupConditionType,
+				Reason:  backupSuccess,
+				Message: fmt.Sprintf("UpgradeBackup for 4.8 -> 4.9 located at path %s/%s on node %q", recentBackupPath, recentBackupName, backupPod.Spec.NodeName),
+			}, nil
+		}
+		return nil, nil
+	case backupPod.Status.Phase == corev1.PodFailed && !isRetryDuration(backupPod.CreationTimestamp.Time, failedPodRetryDuration):
+		return operatorStatusBackupPodFailed("backup pod failed within retry duration: delete skipped"), nil
+	case backupPod.Status.Phase == corev1.PodFailed:
+		// Delete pod
+		err := c.kubeClient.CoreV1().Pods(operatorclient.TargetNamespace).Delete(ctx, backupPod.Name, metav1.DeleteOptions{})
+		if err != nil {
+			return operatorStatusBackupPodFailed(fmt.Sprintf("failed to delete pod openshift-etcd/%s: %v", clusterBackupPodName, err)), nil
+		}
+		return operatorStatusBackupPodFailed("failed pod deletion: successful"), nil
+	}
+
+	return &operatorv1.OperatorCondition{
+		Status:  operatorv1.ConditionUnknown,
+		Type:    backupConditionType,
+		Reason:  "UpgradeBackupInProgress",
+		Message: fmt.Sprintf("Backup pod phase: %q", backupPod.Status.Phase),
+	}, nil
+}
+
+// getBackupNodeName checks all etcd pods and verifies the etcd member on that
+// pod is healthy if yes returns the node name it is scheduled on and errors if
+// no healthy members exist.
+func (c *UpgradeBackupController) getBackupNodeName() (string, error) {
+	pods, err := c.podLister.List(labels.Set{"app": "etcd"}.AsSelector())
+	if err != nil {
+		return "", err
+	}
+
+	var errs []error
+	for _, pod := range pods {
+		if !strings.HasPrefix(pod.Name, "etcd-") {
+			continue
+		}
+		member, err := c.etcdClient.GetMember(pod.Spec.NodeName)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		isHealthy, err := c.etcdClient.IsMemberHealthy(member)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		if isHealthy {
+			return pod.Spec.NodeName, nil
+		}
+	}
+
+	if len(errs) > 0 {
+		return "", utilerrors.NewAggregate(errs)
+	}
+
+	return "", ceohelpers.ErrNodeNotFound
+}
+
+func createBackupPod(ctx context.Context, nodeName, recentBackupName, operatorImagePullSpec, targetImagePullSpec string, client coreclientv1.PodsGetter, recorder events.Recorder) error {
+	pod := resourceread.ReadPodV1OrDie(etcd_assets.MustAsset("etcd/cluster-backup-pod.yaml"))
```

Comment:
Do we really want to die here, I would prefer handling this gracefully as backup is not as critical as having a functioning cluster.

I could be misunderstanding what Die does.

Reply:
it will die if the pod yaml is invalid or does not exist.